src/components/Modal/FormModal.tsx

```diff
@@ -1,38 +1,122 @@
 import { Box } from '@mui/system'
 import Button from '@mui/material/Button'
+import {Wine, WineData} from '../../interfaces/wine_interface'
 
 export const FormModal: React.FC<{
   openAddModal: boolean,
   handleCloseAddModal: () => void
   header: string
-}> = ({ openAddModal, handleCloseAddModal, header }) => {
+  createWine: (wineData: WineData) => Promise<any>
+  updateWinesList: (wine: Wine) => void
+}> = ({ openAddModal, handleCloseAddModal, header, createWine, updateWinesList }) => {
+
+  const [wineValues, setWineValues] = useState<WineData>({
+    name: "",
+    description: "",
+    rating: 0,
+    quantity: 0
+  })
+
+  const [message, setMessage] = useState("")
+  const [errors, setErrors] = useState<string[]>([])
+
+  const onChange: React.ChangeEventHandler<HTMLInputElement> = e => {
+    setWineValues({...wineValues, [e.target.name]: e.target.value})
+  }
+
+  const handleSubmit = async (e: React.SyntheticEvent<HTMLFormElement>) => {
+    e.preventDefault()
+    await createWine(wineValues).then(response => {
+      updateWinesList(response.data.attributes)
+      setWineValues({ name: "", description: "", rating: 0, quantity: 0 })
+      setMessage("Successfully added a wine!")
+    })
+      .catch(error => {
+        setMessage("")
+        const errorData = error.response.data.errors
+        const formattedErrors = formatErrors(errorData)
+        setErrors(formattedErrors)
+    })
+  }
+
+  const formatErrors = (errors: any) => {
+    const errorKeys = Object.keys(errors)
+
+    const formattedErrors = errorKeys.map((errorKey: string) => `${errorKey} ${errors[errorKey]}`)
+
+    return formattedErrors
+  }
+
+  const renderErrors = () => errors.map((error) => <li>{ error }</li>)
+
+  const handleCloseModal = () => {
+    setWineValues({ name: "", description: "", rating: 0, quantity: 0 })
+    setMessage("")
+    setErrors([])
+    handleCloseAddModal()
+  }
 
   return (
-    <Modal data-testid="add-modal-id" open={openAddModal} onClose={handleCloseAddModal}>
+    <Modal data-testid="add-modal-id" open={openAddModal} onClose={handleCloseModal}>
       <Box className="modal-box">
-        <h2>{header}</h2>
+        <div className='form-head-section'>
+          <h2 className="form-header">{header}</h2>
+          {message ? (<p data-testid='submit-success-id'>{message}</p>) : null}
+          {errors ? (<ol className='error-list' data-testid='submit-error-id'>{renderErrors()}</ol>) : null}
+        </div>
         <div className='form'>
-          <form data-testid="form-id">
+          <form data-testid="form-id" onSubmit={handleSubmit}>
+            <div className='form-row'>
+              <label className='form-label'>Name:</label>
+              <input
+                type='text'
+                placeholder='Enter your name here'
+                name="name"
+                value={wineValues.name}
+                onChange={onChange}
+                data-testid='form-name-id' />
```

Comment:
Same comment from last PR: I believe adding a `htmlFor` can make this input/label more accessible. 

https://github.com/skp96/Wine-O-Inventory-Client/pull/3#discussion_r798703645